lib/bolt_server/transport_app.rb

```diff
@@ -45,9 +45,30 @@ def initialize(config)
 
       @file_cache = BoltServer::FileCache.new(@config).setup
 
+      # This is needed until the PAL is threadsafe.
+      @pal_mutex = Mutex.new
+      # These are only needed for the `/plans` endpoint.
+      require 'puppet'
+      require 'bolt/pal'
+      require 'bolt_server/pe/pe_pal'
+
       super(nil)
     end
 
+    get '/plans' do
+      @pal_mutex.synchronize do
+        environment = params['environment']
+        if environment.nil?
+          pal = Bolt::PAL.new(['/opt/puppetlabs/puppet/modules'], nil, nil)
```

Comment:
I think we should make `envirnoment` required and error if it is not specified. I don't think we should implement this with Bolt's pal (also the modulepath for in the init method does not make sense).